components/dashboard/src/projects/Project.tsx

```diff
@@ -15,6 +15,9 @@ import { TeamsContext, getCurrentTeam } from "../teams/teams-context";
 import { prebuildStatusIcon, prebuildStatusLabel } from "./Prebuilds";
 import { ContextMenuEntry } from "../components/ContextMenu";
 import { shortCommitMessage } from "./render-utils";
+import { ErrorCodes } from "@gitpod/gitpod-protocol/lib/messaging/error";
+import { openAuthorizeWindow } from "../provider-utils";
+import ConfirmationModal from "../components/ConfirmationModal";
```

Comment:
**thought**: Instead of a disruptin modal we could use a softer auth flow using a placeholder in the Branches list. See also [relevant discussion]() (internal). WDYT?

| Auth flow for accessing branch information | 
|-|
| <img width="1440" alt="Bracnhes" src="https://user-images.githubusercontent.com/120486/132874668-5f6c7a2b-4821-4965-855e-dbf8fc761687.png"> |

Reply:
Awesome!